python/cudf/cudf/utils/docutils.py

```diff
@@ -335,3 +345,28 @@ def wrapper(func):
         max           <NA>     3.0
 """
 )
+
+doc_reset_index_template = """
```

Comment:
Can we move this to IndexedFrame?

Reply:
I was a bit hesitant about this because I don't want DataFrame/Series to import anything from `IndexedFrame` from our layered architecture. But it looks like we are already importing it, so I guess it's ok.

Reply:
I'm not sure why you wouldn't want DataFrame/Series to use things in the IndexedFrame namespace given that they inherit from the class. Isn't that exactly how the layered architecture should work? You're basically specializing the docstring from the parent in the child, except the method isn't present in the parent so you're just storing the docstring in indexed_frame.py

Reply:
lol, yes sure it's ok. The taboo is not to include outer layer classes in inner layer classes, like including `dataframe` in `indexedframe`. I'm just getting familiar with this so expect errors.

Reply:
Ah yeah sorry I see where your confusion is. Yes, the key is to think of these layers as one-way dependencies. If you have `class B(A)`, `B` knows all about `A` and should be free to specialize any behaviors that `A` doesn't forbid (e.g. private members in C++, in Python you just need some convention to clearly delineate these). What you want is to avoid making `A` depend on behaviors of `B` because that invariably leads to bugs when you get a new `class C(A)` that needs to behave differently from `B`.